### ISSUE TITLE
unable to login if no avatar is provided

### ISSUE THREAD

**Issue by konradmoesch** [nextcloud/user_oidc#904]
I am using user_oidc with Authentik. For the avatar, I have configured a mapping for the avatar to the `avatar` scope (in Authentik, the avatar is also available in the `user.avatar` variable and can be used in a scope mapping)

If no avatar is available in Authentik (I use the default `gravatar,initials` avatar providers setting), the affected user is unable to login. The following errors arise in nextcloud:

```
[PHP] error: Undefined variable $avatarContent at /var/www/nextcloud/apps/user_oidc/lib/Service/ProvisioningService.php#352
	GET /index.php/apps/user_oidc/code?code=XXXXXX
```

```
[index] error: OC_Image::loadFromData(): Argument #1 ($str) must be of type string, null given, called in /var/www/nextcloud/apps/user_oidc/lib/Service/ProvisioningService.php on line 352 in file '/var/www/nextcloud/lib/private/legacy/OC_Image.php' line 806
	GET /index.php/apps/user_oidc/code?code=XXXXXX
```

I believe we should have a check for null/empty before trying to set the avatar in nextcloud, or should have a default / fallback avatar that is set in this case.

**Comment by julien-nc** [comment]
Thanks for reporting this. #905 should fix it. Could you try it?
For that you would need to save `user_oidc/lib/Service/ProvisioningService.php` somewhere and then replace it with this one:
https://raw.githubusercontent.com/nextcloud/user_oidc/e03ddaac78d2240e75aa215bbd29320f11ef5a74/lib/Service/ProvisioningService.php

There were a lot of mistakes in the provisioning code.

**Comment by julien-nc** [comment]
The fix is included in v6.0.1. Could you confirm the issue is fixed and everything is fine on your side? Let's reopen this issue if something is still wrong.

**Comment by konradmoesch** [comment]
Thanks! Now, login without avatar works for me
